src/System/Random/Monad.hs

```diff
@@ -421,11 +424,73 @@ runSTGen_ g action = fst $ runSTGen g action
 --     @UniformRange (a, b)@ instance based on just @UniformRange a@ and
 --     @UniformRange b@.
 
-
 -------------------------------------------------------------------------------
 -- Notes
 -------------------------------------------------------------------------------
 
+-- $floating
+--
+-- The 'UniformRange' instances for 'Float' and 'Double' use the following
+-- procedure to generate a random value in a range for @uniformRM (a, b) g@:
+--
+-- 1.  Generate \(x\) uniformly such that \(0 \leq x \leq 1\).
+--
+--     The method by which \(x\) is sampled does not cover all representable
+--     floating point numbers in the unit interval. The method never generates
+--     denormal floating point numbers, for example.
+--
+-- 2.  Return \((b - a) * x + a\).
+--
+--     Due to rounding errors, floating point operations are neither
+--     commutative, nor associative, nor distributive the way the corresponding
+--     operations on real numbers are. Additionally, floating point numbers
+--     admit special values @NaN@ as well as negative and positive infinity.
```

Comment:
I think because you talk about `NaN` and `Infinity` here it will be better to move this part:
```haskell
-- >>> nan = read "NaN" :: Float
-- >>> inf = read "Infinity" :: Float
--
```
right here as well instead of in the setup, which otherwise  will not be visible in the haddock. Reason why I am suggesting this is because examples below use `nan` and `inf` which are not some common Haskell bindings and it is impossible to see where they come from.

Reply:
Done, thanks!